coding-challenges.js

```diff
@@ -1,60 +1,118 @@
 // ASSESSMENT 4: JavaScript Coding Practical Questions
 
-// --------------------1) Create a function that takes in an array. Each time the function is run, remove the first item from the array and shuffle the remaining content. If the array is empty, return "The array is empty."
-
-
-
-
-
-var collections = ["purple", "blue", "green", "yellow", "pink"]
+// --------------------
+// // 1) Create a function that takes in an array. 
+// // Each time the function is run, 
+// // remove the first item from the array and 
+// shuffle the remaining content. 
+// If the array is empty, return "The array is empty."
+
+var collections = ["purple", "blue", "green", "yellow", "pink"];
+
+function arrMixer(array) {
+   
+    for (let i = array.length - 1; i > 0; i--) {
+        const newI = Math.floor(Math.random() * (i + 1));
+        [array[i], array[newI]] = [array[newI], array[i]];
+    }
+    if (array.shift() >= 0){
+    }
+ return array;
+ // else {
+    // return 'empty array';
+    // }
+}
+
+console.log(arrMixer(collections));
+console.log(arrMixer(collections));   
+console.log(arrMixer(collections));   
+console.log(arrMixer(collections));  
+ 
 // Expected output example1 (can be a different order): ["yellow", "blue", "pink", "green"]
 // Expected output example2 (can be a different order): ["blue", "green", "pink"]
 // Expected output example3 (can be a different order): ["pink", "green"]
 
 
-// --------------------2) Create a function that takes in an array of numbers and returns the sum of all the numbers cubed.
-
-
-
-
-
-var cubeAndSum1 = [2, 3, 4]
+// // --------------------
+// 2) Create a function that 
+// takes in an array of numbers and 
+// returns the sum of all the numbers cubed.
+var cubeAndSum1 = [2, 3, 4];
 // Expected output: 99
-var cubeAndSum2 = [0, 5, 10]
+var cubeAndSum2 = [0, 5, 10];
 // Expected output: 1125
 
+const arrCube = (cube) => {
+    let newCubeArray = 0;
+    for (let i=0; i< cube.length; i++) {
+        newCubeArray = (cube[0]**3+cube[1]**3+cube[2]**3)
+    }
+    return newCubeArray;
+};
 
-
-// --------------------3) Create a function that takes an array of numbers and returns an array of the minimum and maximum numbers in that order.
-
+console.log(arrCube(cubeAndSum1));   
+console.log(arrCube(cubeAndSum2));
 
 
-
-
-var nums1 = [3, 56, 90, -8, 0, 23, 6]
+// --------------------
+// 3) Create a function that 
+// takes an array of numbers and 
+// returns an array of the minimum and maximum numbers in that order.
+var nums1 = [3, 56, 90, -8, 0, 23, 6];
 // Expected output: [-8, 90]
-var nums2 = [109, 5, 9, -59, 8, 24]
+var nums2 = [109, 5, 9, -59, 8, 24];
 // Expected output: [-59, 109]
 
+function arrMinMax(nummm) {
+  return [Math.min(...nummm), Math.max(...nummm)];
+}
```

Comment:
wonderful, concise code. just be ready to write out the long-form in an interview :)